cpp/src/arrow/compute/kernels/row_encoder_internal.h

```diff
@@ -308,27 +468,12 @@ struct ListFixedWidthChildEncoder : KeyEncoder {
       std::static_pointer_cast<T>(type_)->value_type();
     const int32_t child_width = child_type->byte_width();
 
-    if (data.is_array()) {
-      int64_t i = 0;
-      IterateListArray<T>(
-        data.array,
-        [&](Offset list_begin, Offset list_end, const uint8_t*, const uint8_t*) {
-          Offset list_length = list_end - list_begin;
-          lengths[i++] += kExtraByteForNull + sizeof(Offset) +
-            list_length * (child_width + 1);
-        },
-        [&] { lengths[i++] += kExtraByteForNull + sizeof(Offset); });
-    } else {
-      int32_t buffer_size = 0;
-      ProcessListScalar<T>(
-        data.scalar_as<ScalarType>(),
-        [&](int64_t list_length, const uint8_t*, const uint8_t*) {
-          buffer_size = list_length * (child_width + 1);
-        });
-      for (int64_t i = 0; i < batch_length; i++) {
-        lengths[i] += kExtraByteForNull + sizeof(Offset) + buffer_size;
-      }
-    }
+    auto get_list_buffer_size = [&](Offset list_begin, Offset list_end,
```

Comment:
can we make this a static helper function on the encoder class to avoid duplicating this in `Encode`? basically mirroring what you did for the var length list encoder

Reply:
It cannot be static as we do not know what `child_width` is. 

I will bring it out and apply a wrapper like before.